go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -483,6 +519,23 @@ func (n *NodeController) ensureHybridOverlayBridge() error {
 			"stderr: %q, error: %v", stderr, err)
 	}
 
+	// Handle ARP for hybrid external gateway
+	_, _, err = util.RunOVSOfctl("add-flow", extBridgeName,
+		fmt.Sprintf("table=0,priority=100,arp,in_port=%s,arp_tpa=%s,"+
+			"actions=move:NXM_OF_ETH_SRC[]->NXM_OF_ETH_DST[],"+
+			"mod_dl_src:%s,"+
+			"load:0x2->NXM_OF_ARP_OP[],"+
+			"move:NXM_NX_ARP_SHA[]->NXM_NX_ARP_THA[],"+
+			"load:0x%s->NXM_NX_ARP_SHA[],"+
+			"move:NXM_OF_ARP_TPA[]->NXM_NX_REG0[],"+
+			"move:NXM_OF_ARP_SPA[]->NXM_OF_ARP_TPA[],"+
+			"move:NXM_NX_REG0[]->NXM_OF_ARP_SPA[],"+
+			"IN_PORT",
```

Comment:
Need vxlan header here:
ovs-ofctl add-flow br-ext 'priority=5000,arp,in_port="ext-vxlan",arp_tpa=172.10.6.0/23,actions=move:NXM_OF_ETH_SRC[]->NXM_OF_ETH_DST[],mod_dl_src:0a:58:ac:0a:06:03,load:0x2->NXM_OF_ARP_OP[],move:NXM_NX_ARP_SHA[]->NXM_NX_ARP_THA[],load:0xa58ac0a0603->NXM_NX_ARP_SHA[],move:NXM_OF_ARP_TPA[]->NXM_NX_REG0[],move:NXM_OF_ARP_SPA[]->NXM_OF_ARP_TPA[],move:NXM_NX_REG0[]->NXM_OF_ARP_SPA[],load:0x1001->NXM_NX_TUN_ID[0..31],load:0xa4b4796->NXM_NX_TUN_IPV4_DST[],IN_PORT'

Reply:
so this means we need namespaceExternalGw annotation here.
Then I think we need to move this flow to addOrUpdatePod, it's no longer static.

Agree @trozer @dcbw ?

Reply:
Or something like :+1: 
```
			"actions=move:tun_src->tun_dst",
			"move:NXM_OF_ETH_SRC[]->NXM_OF_ETH_DST[],"+
			"mod_dl_src:%s,"+
```

Reply:
done @dcbw